xpk.py

```diff
@@ -2481,14 +2481,37 @@ def workload_delete(args) -> int:
   if set_cluster_command_code != 0:
     xpk_exit(set_cluster_command_code)
 
-  yml_string = workload_delete_yaml.format(args=args)
-  tmp = write_temporary_file(yml_string)
-  command = f'kubectl delete -f {str(tmp.file.name)}'
-  return_code = run_command_with_updates(command, 'Delete Workload', args)
-
-  if return_code != 0:
-    xpk_print(f'Delete Workload request returned ERROR {return_code}')
-    xpk_exit(return_code)
+  if not args.workload:
+    args.filter_by_status = "EVERYTHING"
+    columns = {
+      'Jobset Name': '.metadata.ownerReferences[0].name',
+    }
+    xpk_print("Get the name of the workloads in the cluster.")
+    return_code, return_value = get_workload_list(args, columns)
+
+    if return_code != 0:
+      xpk_print(f'List Job request returned ERROR {return_code}')
+      xpk_exit(return_code)
+    # Skip the header
+    workloads = return_value.strip().split("\n")[1:]
+  else:
+    workloads = [args.workload]
+
+  if not workloads:
+    xpk_print("There are no workloads to delete matching the filter in the cluster.")
+
+  for workload in workloads:
+    result = input(f'Do you want to delete the workload: {workload} (Y/n)? ')
```

Comment:
I like the yes / no user feedback logic here! two suggestions:

1. Can the user enter yes/no/y/n once per workload delete command, instead of once per workload. We can print out all the workloads that are planned to be deleted. (Same as https://github.com/google/xpk/blob/main/xpk.py#L1396-L1401)

I think it would be hard to ask the user to type yes several times. 

2. Can you refactor the user input y/yes/n/no logic we use in node pool delete into a function and use that here as well. I think we will also need it in cluster deletion as well for a new feature request. (https://github.com/google/xpk/blob/main/xpk.py#L1396-L1401)

Reply:
Updated!